tikv-client/src/main/java/com/pingcap/tikv/region/TiRegion.java

```diff
@@ -124,8 +142,10 @@ boolean switchPeer(long leaderStoreID) {
   }
 
   public boolean contains(ByteString key) {
-    return meta.getStartKey().equals(key)
-        && (meta.getEndKey().equals(key) || meta.getEndKey().isEmpty());
+    return (meta.getStartKey().isEmpty() || FastByteComparisons.compareTo(meta.getStartKey().toByteArray(),
```

Comment:
omg this line is not what it should be.
If you are not using this line, you may simply delete it.
A correct version should be: using `getRegionByKey` to identify if it is the same region with what you are asking for.

Reply:
No, ZheXuan says it should be what I write